HW_6/task_2.js

```diff
@@ -0,0 +1,271 @@
+          employees_count: 10,
+        },
+        {
+          id: 3,
+          name: "Отдел маркетинга",
+          employees_count: 20,
+        },
+        {
+          id: 4,
+          name: "Администрация",
+          employees_count: 15,
+        },
+      ]
+    },
+    {
+      id: 5,
+      name: "Предприятие 2",
+      departments: [
+        {
+          id: 6,
+          name: "Отдел разработки",
+          employees_count: 50,
+        },
+        {
+          id: 7,
+          name: "Отдел маркетинга",
+          employees_count: 20,
+        },
+        {
+          id: 8,
+          name: "Отдел охраны труда",
+          employees_count: 5,
+        },
+      ]
+    },
+    {
+      id: 9,
+      name: "Предприятие 3",
+      departments: [
+        {
+          id: 10,
+          name: "Отдел аналитики",
+          employees_count: 0,
+        },
+      ]
+    }
+  ];
+
+  function task1(obj) {
+    obj.forEach(firm => {  
+        const totalEmployees = sumEmployees(firm.departments);              
+        console.log(`${firm.name} (${totalEmployees})`);
+        firm.departments.forEach(department => {
+            const employeesWord = getEmployeesWord(department.employees_count);
+            console.log(`  ${department.name} (${department.employees_count} ${employeesWord})`);
+        });
+    });
+};
+
+
+function sumEmployees(departments) {
+    return departments.reduce((sum, department) => sum + department.employees_count, 0);
+};
+
+
+function getEmployeesWord(count) {
+    if (count % 10 === 1 && count % 100 !== 11) {
+        return 'сотрудник';
+    } else if ([2, 3, 4].includes(count % 10) && ![12, 13, 14].includes(count % 100)) {
+        return 'сотрудника';
+    } else {
+        return 'сотрудников';
+    }
+};
+
+task1(enterprises);
+
+
+
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+function getEnterpriseName(identifier) {
+    let result = null;
+
+    enterprises.find(enterprise => {
+        if (enterprise.id === identifier) {
+            result = enterprise.name;
+            return true;
+        };
+
+        const department = enterprise.departments.find(department => department.name === identifier);
+        if (department) {
+            result = enterprise.name;
+            return true;
+        };
+
+        return false;
+    });
+
+    return result;
+};
+
+console.log(getEnterpriseName(1));
+console.log(getEnterpriseName("Отдел маркетинга"))
+
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+// Пример:
+// addEnterprise("Название нового предприятия")
+
+function addEnterprise(nameEnterprise) {    
+    const maxId = enterprises.reduce((max, enterprise) => Math.max(max, enterprise.id), 0);    

```

Comment:
1. getMaxId should be used as a helper function 
2. It should calculate new id not depending only in enterprises ids, but should search for the greatest one among all enterprises and departments